[DOCS-6190] Add quickstart to Obs Pipelines setup doc (#19688)
DataDog/documentation: content/en/observability_pipelines/setup/_index.md

aliases:
  - /getting_started/observability_pipelines/
  - /observability_pipelines/installation/
further_reading:
  - link: "/observability_pipelines/working_with_data/"
    tag: "Documentation"
    text: "Working with data in Observability Pipelines"
  - link: "/observability_pipelines/production_deployment_overview/"
    tag: "Documentation"
    text: "Deployment Design and Principles for the Observability Pipelines Worker"
  - link: "/observability_pipelines/architecture/"
    tag: "Documentation"
    text: "Production deployment design and principles for the Observability Pipelines Worker"
  - link: "https://dtdg.co/d22op"
    tag: "Learning Center"
    text: "Safe and Secure Local Processing with Observability Pipelines"
---

## Overview

The [Observability Pipelines Worker][1] can collect, process, and route logs and metrics from any source to any destination. Using Datadog, you can build and manage all of your Observability Pipelines Worker deployments at scale.

There are several ways to get started with the Observability Pipelines Worker.

> - [Quickstart](#quickstart): Install the Worker and test out the functionality of Observability Pipelines with demo data.

Reviewer: Something like "Quickstart: Install the Worker with a simple pipeline that emits demo data to get started quickly" is better than explicitly saying testing out the functionality or locally test, since experienced users will use the Quickstart to easily get a new pipeline going

- [Datadog setup guide][2]: Install the Worker and connect it to Datadog.

> - [Splunk setup guide][3]: Install the Worker and connect it to Splunk.

Reviewer: "Splunk setup guide: Install the Worker with an out-of-the-box pipeline for receiving and routing data from Splunk HEC to both Splunk and Datadog"

This document walks you through the quickstart installation steps and then provides resources for next steps.

## Prerequisites

To install the Observability Pipelines Worker, you need the following:

- A valid [Datadog API key][4].
- A pipeline ID.

To generate a new API key and pipeline:

1. Navigate to [Observability Pipelines][5].
2. Click **New Pipeline**.
3. Enter a name for your pipeline.
4. Click **Next**.
4. Select the template you want and follow the instructions.

## Quickstart

Follow the below instructions to install the Worker and deploy a sample pipeline configuration that uses demo data.

### Install the Observability Pipelines Worker

{{< tabs >}}
{{% tab "Docker" %}}

The Observability Pipelines Worker Docker image is published to Docker Hub [here][1].

> 1. Download the [sample pipeline configuration file][2] that uses demo data for this quickstart. See [Configurations][3] for more information about the source, transform, and sink used in the sample configuration.

Reviewer: "1. Download the [sample pipeline configuration file][2] to quickly get started. This configuration will emit demo data, parse and structure the data, and emit them to console and Datadog. See [Configurations][3] for more information about the source, transform, and sink used in the sample configuration."

2. Run the following command to start the Observability Pipelines Worker with Docker:

>     ```

Reviewer 2: ```suggestion
    
    ```
```
I think there's a small issue here with the formatting that is rendering the rest of this document oddly in the markdown. It doesn't seem to be impacting how it's rendering on the site, but it is still probably better to fix it to prevent future issues.

My guess is that it's the spacing between the ordered list and the codeblock.

    docker run -i -e DD_API_KEY=<API_KEY> \
      -e DD_OP_PIPELINE_ID=<PIPELINE_ID> \
      -e DD_SITE=<SITE> \
      -p 8282:8282 \
      -v ./pipeline.yaml:/etc/observability-pipelines-worker/pipeline.yaml:ro \
      datadog/observability-pipelines-worker run
    ```

>     Replace `<API_KEY>` with your Datadog API key, `<PIPELINES_ID>` with your Observability Pipelines configuration ID, and `<SITE>` with {{< region-param key="dd_site" code="true" >}}. `./pipeline.yaml` must be the relative or absolute path to the configuration you downloaded in Step 1.

Reviewer: Can we add "To note, " at the head of "`./pipeline.yaml` must be the relative or absolute path to the configuration you downloaded in Step 1." sentence?

    ```shell
    helm repo add datadog https://helm.datadoghq.com
    ```
    ```shell
    helm repo update
    ```
    ```shell
    helm upgrade --install \
      opw datadog/observability-pipelines-worker \
      -f google_gke.yaml
    ```

[1]: /resources/yaml/observability_pipelines/quickstart/google_gke.yaml
[2]: /observability_pipelines/configurations/
{{% /tab %}}
{{% tab "APT-based Linux" %}}

Install the Worker with the one-line install script or manually.
#### One-line installation script

1. Run the one-line install command to install the Worker. Replace `<DD_API_KEY>` with your Datadog API key, `<PIPELINES_ID>` with your Observability Pipelines ID, and `<SITE>` with {{< region-param key="dd_site" code="true" >}}.

    ```
    DD_API_KEY=<DD_API_KEY> DD_OP_PIPELINE_ID=<PIPELINES_ID> DD_SITE=<SITE> bash -c "$(curl -L https://s3.amazonaws.com/dd-agent/scripts/install_script_op_worker1.sh)"
    ```

2. Download the [sample configuration file][1] to `/etc/observability-pipelines-worker/pipeline.yaml` on the host. See [Configurations][2] for more information about the source, transform, and sink used in the sample configuration.

3. Start the worker:

>     ```

Reviewer 2: ```suggestion

    ```
```

    curl https://keys.datadoghq.com/DATADOG_APT_KEY_CURRENT.public | sudo gpg --no-default-keyring --keyring /usr/share/keyrings/datadog-archive-keyring.gpg --import --batch
    curl https://keys.datadoghq.com/DATADOG_APT_KEY_F14F620E.public | sudo gpg --no-default-keyring --keyring /usr/share/keyrings/datadog-archive-keyring.gpg --import --batch
    curl https://keys.datadoghq.com/DATADOG_APT_KEY_C0962C7D.public | sudo gpg --no-default-keyring --keyring /usr/share/keyrings/datadog-archive-keyring.gpg --import --batch
    ```

3. Run the following commands to update your local `apt` repo and install the Worker:

    ```
    sudo apt-get update
    sudo apt-get install observability-pipelines-worker datadog-signing-keys
    ```

4. Add your keys and the site ({{< region-param key="dd_site" code="true" >}}) to the Worker's environment variables:

    ```
    sudo cat <<-EOF > /etc/default/observability-pipelines-worker
    DD_API_KEY=<API_KEY>
    DD_OP_PIPELINE_ID=<PIPELINE_ID>
    DD_SITE=<SITE>
    EOF
    ```

5. Download the [sample configuration file][1] to `/etc/observability-pipelines-worker/pipeline.yaml` on the host.

> 6. Start the worker:

Reviewer 2: ```suggestion
6. Start the Worker:
```

>     ```

Reviewer 2: ```suggestion
    
    ```
```

    sudo systemctl restart observability-pipelines-worker
    ```

[1]: /resources/yaml/observability_pipelines/datadog/pipeline.yaml
[2]: /observability_pipelines/configurations/
{{% /tab %}}
{{% tab "RPM-based Linux" %}}

Install the Worker with the one-line install script or manually.

#### One-line installation script

1. Run the one-line install command to install the Worker. Replace `<DD_API_KEY>` with your Datadog API key, `<PIPELINES_ID>` with your Observability Pipelines ID, and `<SITE>` with {{< region-param key="dd_site" code="true" >}}.

    ```
    DD_API_KEY=<DD_API_KEY> DD_OP_PIPELINE_ID=<PIPELINES_ID> DD_SITE=<SITE> bash -c "$(curl -L https://s3.amazonaws.com/dd-agent/scripts/install_script_op_worker1.sh)"
    ```

2. Download the [sample configuration file][1] to `/etc/observability-pipelines-worker/pipeline.yaml` on the host. See [Configurations][2] for more information about the source, transform, and sink used in the sample configuration.

3. Run the following command to start the Worker:

>     ```

Reviewer 2: ```suggestion
    
    ```
```

  metrics_add_dd_tags:
    type: remap
    inputs:
      - datadog_agent.metrics
    source: |
      .tags.sender = "observability_pipelines_worker"
      .tags.opw_aggregator = get_hostname!()

## This buffer configuration is split into the following, totaling the 288GB
## provisioned automatically by the Terraform module:
## - 240GB buffer for logs
## - 48GB buffer for metrics
##
## This should work for the vast majority of OP Worker deployments and should rarely
## need to be adjusted. If you do change it, be sure to update the `ebs-drive-size-gb`
## parameter.
sinks:
  datadog_logs:
    type: datadog_logs
    inputs:
      - logs_finish_ddtags
    default_api_key: "$${DD_API_KEY}"
    compression: gzip
    buffer:
       type: disk
       max_size: 257698037760
  datadog_metrics:
    type: datadog_metrics
    inputs:
      - metrics_add_dd_tags
    default_api_key: "$${DD_API_KEY}"
    buffer:
      type: disk
      max_size: 51539607552
EOT
}
```

[2]: /observability_pipelines/configurations/
{{% /tab %}}
{{< /tabs >}}

See [Working with Data][6] for more information on transforming your data.

## Next steps

> The quickstart walked you through how to install the Worker and deploy a sample pipeline configuration. For instructions on how to install the Worker to receive and route data from your Datadog Agents to Datadog or to receive and route data from your Splunk HEC to Splunk and Datadog, select your specific use case:

Reviewer 2: ```suggestion
The quickstart walked you through installing the Worker and deploying a sample pipeline configuration. For instructions on how to install the Worker to receive and route data from your Datadog Agents to Datadog or to receive and route data from your Splunk HEC to Splunk and Datadog, select your specific use case:
```
